### PR TITLE
Refine on Recursive ACL.

### PR DESCRIPTION
**Pull request by katmsft**
Resubmitting #1470, original PR is corrupted and CI behavior is strange.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@
 - Added `Metadata`, `AccessType`, `HasImmutabilityPolicy`, `HasLegalHold`, `LeaseDuration`, `LeaseState` and `LeaseStatus` to `FileSystemItem`.
 - Added new type `LeaseDurationType` to indicate if a lease duration is fixed or infinite.
 - Added `RequestId` in each return type for REST API calls, except for concurrent APIs.
+- Added `UpdateAccessControlRecursiveSinglePage` to update the access control recursively for a datalake directory.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:18]:
```suggestion
- Added `UpdateAccessControlListRecursiveSinglePage` to update the access control recursively for a datalake directory.
```

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,8 @@
...
+- Added `RemoveAccessControlRecursiveSinglePage` to remove the access control recursively for a datalake directory.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:19]:
```suggestion
- Added `RemoveAccessControlListRecursiveSinglePage` to remove the access control recursively for a datalake directory.
```

**Comment by Jinming-Hu** [comment]
We should move `SetAccessControlListRecursive` functions to `PathClient`

**Comment by katmsft** [comment]
> We should move `SetAccessControlListRecursive` functions to `PathClient`

I don't think it makes sense for File client, so we shouldn't.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp [hunk 3]
```diff
@@ -204,6 +204,69 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+     * @param acls Sets POSIX access control rights on files and directories. Each access control
+     * entry (ACE) consists of a scope, a type, a user or group identifier, and permissions.
+     * @param options Optional parameters to set an access control recursively to the resource the
+     * directory points to.
+     * @return
+     * Azure::Core::Response<Models::SetDataLakePathAccessControlRecursiveSinglePageResult>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::SetDataLakePathAccessControlRecursiveSinglePageResult>
+    SetAccessControlRecursiveSinglePage(
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:219]:
```suggestion
    SetAccessControlListRecursiveSinglePage(
```

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**